### ISSUE TITLE
Support for optional icons

### ISSUE THREAD

**Issue by hadronized** [vigoux/notifier.nvim#3]
I know it’s not a documented nor default thing, but [nvim-notify](https://github.com/rcarriga/nvim-notify) has this `opts.icon` support that is quite nice, and I think you should also support it (I set it my plugins / configuration / shortcuts so it feels like a feature loss to me). And you already read the title so… :)

**Comment by vigoux** [comment]
I'll look into it, but it seems to be an incremental improvement from the title, so I'll add an initial implementation ASAP

**Comment by lostl1ght** [comment]
Hi, thanks for the plugin.

Is there a reason for this check on line 199?
```lua
 if content.icon and displayw(content.icon) > 1 then
    error("Message icon should be of length 1 character.")
  end
```
It breaks some notifications for me. I have removed these lines and it works just fine.

**Comment by vigoux** [comment]
This check is indeed not needed. 
But the general idea was that an icon is really an icon rather than a full piece of text

**Comment by vigoux** [comment]
Should be fixed by 7e7a83d1a34a1b01014077586c174e033f1cdfa5.

**Comment by lostl1ght** [comment]
It is. Thank you.
